### PR TITLE
executor: migrate test-infra to testify for partition_table_test.go

### LINKED ISSUE #28600: migrate test-infra to testify for `partition_table_test.go`

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: close #28600

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: executor/partition_table_test.go [hunk 1]
```diff
@@ -305,12 +330,12 @@ func (s *partitionTableSuite) TestPartitionInfoDisable(c *C) {
 	tk.MustQuery("select * from t_info_null where (date = '2020-10-02' or date = '2020-10-06') and app = 'xxx' and media = '19003006'").Check(testkit.Rows())
 }
 
-func (s *partitionTableSuite) TestOrderByandLimit(c *C) {
-	if israce.RaceEnabled {
-		c.Skip("exhaustive types test, skip race test")
-	}
```
Comment (tangenta) [inline executor/partition_table_test.go:311]: Do you know why it is skipped in the past?

**Review by tangenta — APPROVED** [review]

**Review by tisonkun — COMMENTED** [review]
unit_test failed on

```
[2021-12-23T06:07:33.363Z] FAIL	github.com/pingcap/tidb/executor	518.475s
```

File: executor/partition_table_test.go (hunk 1 above)
Comment (hawkingrei) [reply executor/partition_table_test.go:311]: I don't know. but I have to fix it.

Comment (hawkingrei) [reply executor/partition_table_test.go:311]: Done, All have been removed.

**Comment by hawkingrei** [comment]
@tisonkun @wshwsh12 PTAL

**Review by tisonkun — APPROVED** [review]

**Pull request merged**